go-controller/pkg/ovn/service.go

```diff
@@ -165,6 +167,130 @@ func (ovn *Controller) syncServices(services []interface{}) {
 	}
 }
 
+func (ovn *Controller) createService(service *kapi.Service) error {
+	klog.V(5).Infof("Creating service %s", service.Name)
+	if !util.IsClusterIPSet(service) {
+		return fmt.Errorf("no cluster IP for service %s found", service.Name)
+	} else if len(service.Spec.Ports) == 0 {
+		return fmt.Errorf("no Ports specified for service")
+	}
+
+	for _, svcPort := range service.Spec.Ports {
+		var port int32
+		if util.ServiceTypeHasNodePort(service) {
+			port = svcPort.NodePort
+		} else {
+			port = svcPort.Port
+		}
+		if port == 0 {
+			continue
+		}
+
+		protocol := svcPort.Protocol
+		if protocol == "" || (protocol != TCP && protocol != UDP) {
+			protocol = TCP
+		}
+
+		if util.ServiceTypeHasNodePort(service) {
+			// Each gateway has a separate load-balancer for N/S traffic
+
+			physicalGateways, _, err := ovn.getOvnGateways()
+			if err != nil {
+				return err
+			}
+
+			for _, physicalGateway := range physicalGateways {
+				loadBalancer, err := ovn.getGatewayLoadBalancer(physicalGateway,
+					protocol)
+				if err != nil {
+					klog.Errorf("physical gateway %s does not have load_balancer "+
+						"(%v)", physicalGateway, err)
+					continue
+				}
+				if loadBalancer == "" {
+					continue
+				}
+				physicalIP, err := ovn.getGatewayPhysicalIP(physicalGateway)
+				if err != nil {
+					klog.Errorf("physical gateway %s does not have physical ip (%v)",
+						physicalGateway, err)
+					continue
+				}
+				// With the physical_ip:port as the VIP, add an entry in
+				// 'load_balancer'.
+				vip := util.JoinHostPortInt32(physicalIP, port)
+				// Skip creating LB if endpoints watcher already did it
+				if eps, ok := ovn.GetServiceLBEndpoints(loadBalancer, vip); ok && len(eps) > 0 {
+					klog.V(5).Infof("Load Balancer already configured for %s, %s", loadBalancer, vip)
+				} else if ovn.svcQualifiesForReject(service) {
+					aclUUID, err := ovn.createLoadBalancerRejectACL(loadBalancer, physicalIP, port, protocol)
+
+					if err != nil {
+						return fmt.Errorf("failed to create service ACL")
+					} else {
+						klog.V(5).Infof("Service Reject ACL created for physical gateway: %s", aclUUID)
+					}
```

Comment:
remove the `else` alarm and just log the message.

Reply:
will remove it